lib/plug/conn.ex

```diff
@@ -653,6 +653,39 @@ defmodule Plug.Conn do
     %{conn | resp_headers: List.keystore(headers, key, 0, {key, value})}
   end
 
+  @doc ~S"""
+  Similar to `put_resp_header` this functions adds a new response header (`key`)
+  but rather then replacing the exising one it appends another header with the same `key`.
+
+  It is recommended for header keys to be in lower-case, to avoid sending
+  duplicate keys in a request. As a convenience, this is validated during
+  testing which raises a `Plug.Conn.InvalidHeaderError` if the header key
+  is not lowercase.
+
+  Raises a `Plug.Conn.AlreadySentError` if the connection has already been
+  `:sent` or `:chunked`.
+
+  Raises a `Plug.Conn.InvalidHeaderError` if the header value contains control
+  feed (`\r`) or newline (`\n`) characters.
+  """
+  def prepend_resp_headers(%Conn{state: :sent}, _headers) do
```

Comment:
I would add `@spec prepend_resp_headers(t, headers) :: t`

Reply:
In addition, should the spec be `@spec prepend_resp_headers(t, headers) :: t | no_return` because it could raise `AlreadySentError` or `InvalidHeaderError`?

Reply:
@fertapric Thanks! I addressed this in ef83dc3. About the `no_return`. I have no answer, but `put_resp_header` also has only `t` as return type.

```elixir
@spec put_resp_header(t, binary, binary) :: t
```

Reply:
We only add no_return for functions that always raise. `t | no_return` is the same as `t`.